frontend/state_management/Slices/QuerySlice.ts

```diff
@@ -0,0 +1,37 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { QueryState } from '../Reducers/QueryReducers';
```

Comment:
Please just move the interface type from the old reducer into your new slice so we can delete old files without issues later.